### PR TITLE
fix(dependencies): trims dependencies and fixes tests.

### PR DESCRIPTION
**Pull request by runspired**
We may want to remove other dependencies as well, because it forces you to need a large number of `resolutions` to avoid a failing Travis build.

### CODE REVIEW

**Comment by runspired** [comment]
It appears @rwjblue introduced this with the `ember-cli` conversion: https://github.com/ember-cli/ember-resolver/commit/2b06d4e7428c93e0b36fa9d0376c718d00716bcd#diff-0a08a7565aba4405282251491979bb6bR4

I'll update this to remove all of the dependencies other than Ember.

**Comment by runspired** [comment]
Only question I have is what the Ember version should be.  Currently it stands at `"ember: "2.0.0"`.  Should this be changed to `2.2.0`?

https://github.com/runspired/ember-resolver/blob/fix-bower-dependency/bower.json#L4

File: tests/unit/container-debug-adapter-test.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 import Ember from 'ember';
-import Resolver from 'ember-resolver/resolver';
+import Resolver from 'ember-resolver';
 import ContainerDebugAdapter from 'ember-resolver/container-debug-adapter';
 import ContainerDebugAdapterInitializer from 'dummy/initializers/container-debug-adapter';
 
 let containerDebugAdapter, App;
 
 var modules = {};
-function def(module) {
```
Comment (runspired) [inline tests/unit/container-debug-adapter-test.js]: using `module` here was colliding with the `import` of `module` from `qunit`.

**Comment by stefanpenner** [comment]
> Currently it stands at "ember: "2.0.0". Should this be changed to 2.2.0?

:+1:

File: bower.json [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
   "name": "ember-resolver",
-  "dependencies": {
-    "ember": "2.0.0",
-    "ember-cli-shims": "ember-cli/ember-cli-shims#0.0.5",
-    "ember-cli-test-loader": "ember-cli-test-loader#0.1.3",
-    "ember-load-initializers": "ember-cli/ember-load-initializers#0.1.5",
-    "ember-qunit": "0.4.10",
-    "ember-qunit-notifications": "0.0.7",
+  "devDependencies": {
+    "ember": "components/ember#release",
```
Comment (runspired) [inline bower.json]: @stefanpenner I took the ember-data approach for the version, should I go back to `2.2.0`? and should I squash this into one commit?

Comment (stefanpenner) [inline bower.json]: > I took the ember-data approach for the version

:+1: 



> and should I squash this into one commit?

Im not picky, your choice :)

**Comment by rwjblue** [comment]
Ember resolver is no longer a bower dep. It is now a proper addon.

**Comment by rwjblue** [comment]
ember-resolver is used from npm for 2.2.0-beta.1 (and ember cli master). This means that the PR generally isn't needed, but I do agree that we should remove ember-resolver from the bower.json here because it is not needed...

**Comment by stefanpenner** [comment]
@rwjblue I believe it does improve the experience for existing non beta users. (Still the majority) although we should double check this doesn't affect our new planned happy path

**Comment by rwjblue** [comment]
@stefanpenner - Master of this repo (or really any of the 2.x versioning) is completely unusable from bower. Making it be possible to install from bower, and still result in a completely non functional experience seems bizarre. 

Also, all versions of ember-cli prior to the current 2.x beta have used 0.1.x, and the changes to an npm package are in 2.x. Even if users had ^0.1.20, they would not be getting this version...

**Comment by runspired** [comment]
@rwjblue I discovered this because I did `bower install --save-dev ember-resolver` to see if I was missing any nice new upgrades.

I've been doing that with a lot ever since realizing that I needed to bump many of them in order to use newer versions of Ember.  While yes, sem-ver should prevent this, the issue is that with ember-cli releases lagging ember releases, knowing which dependencies you need to have to go with an ember version that doesn't ship with ember-cli has gotten a little problematic.  Hopefully this issue goes away soon, as ember-cli seems to have gotten past the bigger blockers and can possibly release closer to ember in the cycle now.

**Comment by runspired** [comment]
Also, regardless of the bower stuff, this fixes test failures.  In the current setup, `module` was always the `module` from qunit, not the string being passed into the method, which seems to me like perhaps it's a bug in `babel`?

**Comment by rwjblue** [comment]
@runspired - The changes here do not actually make master work when used by bower. It doesn't make much sense to me to shuffle the deps around to allow installing without error/warning but still leave a completely non-usable resulting build.

**Comment by rwjblue** [comment]
Agreed on be module change btw, I'm talking about the bower.json changes specifically.

**Comment by runspired** [comment]
@rwjblue I agree with that.  Can we deprecate specific versions on bower the way you can with npm packages?  And stick a warning in that it won't work?

**Comment by runspired** [comment]
After digging around, seems bower doesn't offer a way to deprecate at all :/ (+1 for npm)

We could create a small module that would be loaded if this were installed via bower that throws a warning.

**Comment by runspired** [comment]
e.g. create a small package at `warn/warn-deprecation.js` and have bower point to it as `main`, and when loaded it emits the error.

**Comment by runspired** [comment]
If that approach (load a module for the bower main file that throws the deprecation) is acceptable, I'll add that to this PR. @rwjblue @stefanpenner

**Comment by stefanpenner** [comment]
Sorry for letting this go stale.

**Comment by stefanpenner** [comment]
https://github.com/ember-cli/ember-resolver/pull/172

**Pull request closed without merging**